imap_processing/tests/ialirt/unit/test_process_codicelo.py

```diff
@@ -0,0 +1,104 @@
+import numpy as np
+import pytest
+
+from imap_processing import imap_module_directory
+from imap_processing.cdf.utils import load_cdf
+from imap_processing.ialirt.l0.process_codicelo import (
+    append_cod_lo_data,
+    find_groups,
+    process_codicelo,
+)
+from imap_processing.utils import packet_file_to_datasets
+
+
+@pytest.fixture(scope="session")
+def xtce_codicelo_path():
+    """Returns the xtce directory."""
+    return (
+        imap_module_directory / "ialirt" / "packet_definitions" / "ialirt_codicelo.xml"
+    )
+
+
+@pytest.fixture(scope="session")
+def binary_packet_path():
+    """Returns the xtce directory."""
+    return (
+        imap_module_directory
+        / "tests"
+        / "ialirt"
+        / "test_data"
+        / "l0"
+        / "apid01152.tlm"
+    )
+
+
+@pytest.fixture(scope="session")
+def codicelo_test_data():
+    """Returns the test data directory."""
+    data_path = (
+        imap_module_directory
+        / "tests"
+        / "ialirt"
+        / "test_data"
+        / "l0"
+        / "imap_codice_l1a_lo-ialirt_20241110193700_v0.0.0.cdf"
+    )
+    data = load_cdf(data_path)
+
+    return data
+
+
+@pytest.fixture()
+def xarray_data(binary_packet_path, xtce_codicelo_path):
```

Comment:
```suggestion
def codicelo_test_data(binary_packet_path, xtce_codicelo_path):
```

For consistency with my logic from the above comment